### DISCUSSION TITLE
Turtle tools durability

### DISCUSSION THREAD

**Discussion by vico93** [cc-tweaked/CC-Tweaked#1820] [General]
Just checked out [the new website](https://datapacks.madefor.cc/) to generate datapacks that change CC:T behavior, and noticed in the hints of the examples that now turtles can consume the durability of equipped tools, right?

If positive, was the diamond toolset behavior changed (now consuming durability), and if not, is possible to override their config to consume (at same time making netherite toolset not consume)?

**Comment by SquidDev** [comment]
The default behaviour for diamond tools has not changed. But yes, you can override those upgrades with a datapack and add `"consumesDurability": "always"` to make them always consume durability.

**Reply by vico93** [reply]
Oh i see. And to make the tool *not* consume durability is putting `never` or just ommit this line?

**Reply by SquidDev** [reply]
Yep.

**Reply by vico93** [reply]
Last question (i hope): there is the `when_enchanted` argument for `consumeDurability` that, i think, only make enchanted tools degrade when used. Is there the opposite option? Degrade only unenchanted tools?

**Reply by SquidDev** [reply]
There is not, no!

Apologies if you're aware already, but the Javadoc details all available options (on the [tool](https://tweaked.cc/mc-1.20.x/javadoc/dan200/computercraft/api/turtle/TurtleUpgradeDataProvider.ToolBuilder.html), and [for the `consumeDurability` option](https://tweaked.cc/mc-1.20.x/javadoc/dan200/computercraft/api/turtle/TurtleToolDurability.html)). Just hopefully be a bit more clear than guessing from the tool JSON.

**Reply by vico93** [reply]
Oh sorry, didn't know those properties were on the javadoc.
